plugins/genocide/genocide.cpp

```diff
@@ -16,29 +16,47 @@ class genocide : public bz_Plugin
     virtual void Cleanup ( void );
     virtual void Event ( bz_EventData *eventData );
 
-    bool allowSuicide = false;
-    bool rogueGenocide = false;
+    bool disableSuicide = false;
+    bool rogueAsTeam = false;
 };
 
 BZ_PLUGIN(genocide)
 
-void genocide::Init ( const char* commandLine )
+void genocide::Init ( const char* cmdLine )
 {
     bz_debugMessage(4, "genocide plugin loaded");
-    if (commandLine)
+    if (strlen(cmdLine) != 0)
     {
-        int conf = atoi(commandLine);
-        if (conf == 1 || conf == 2)
-            rogueGenocide = true;
-        if (conf == 2)
-            allowSuicide = true;
+        if (strcmp(cmdLine, "rogueAsTeam,disableSuicide") == 0)
```

Comment:
this class of methods are dangerous. Prefer `strncmp` and friends instead. Or, better yet, use `std::string` instead.